src/module_wrap.cc

```diff
@@ -173,9 +177,11 @@ void ModuleWrap::Link(const FunctionCallbackInfo<Value>& args) {
     }
     Local<Promise> resolve_promise = resolve_return_value.As<Promise>();
     obj->resolve_cache_[specifier_std].Reset(env->isolate(), resolve_promise);
+
+    promises->Set(mod_context, specifier, resolve_promise).FromJust();
```

Comment:
This line does not set array items (i.e. it does not set `promises[0]`, `promises[1]`, etc.) It sets `promises[specifier]` i.e. some non-array properties of `promises` object. So, the array returned from this function is not populated properly and the corresponding `await SafePromise.all(promises)` does not do what was expected. There should be `i` instead of `specifier`, i.e., `promises->Set(mod_context, i, resolve_promise).FromJust();`, right?

Reply:
yea i actually just caught this while testing. i'm kinda sad this pr landed without tests, as it doesn't work. i have another pr incoming anyway.

Reply:
Should've caught this while reviewing... my bad as well.